### ISSUE TITLE
0.5.0 protects/hides useful properties

### ISSUE THREAD

**Issue by jwag956** [maxcountryman/flask-login#471]
**Describe the bug**
As part of #378 and now in release 0.5.0 - request_callback and user_callback were made 'protected' (by adding a leading underscore). These are valuable properties for upstream packages (such as Flask-Security) so that applications can override the login_manager and provide their own implementations, and still have middleware be able to call with a uniform API.
It isn't clear why those were modified as part of that PR.

**Expected behavior**
Make those 2 publicly accessible methods or properties.

**Comment by alanhamlett** [comment]
Those properties aren't meant to be directly used normally. They should be used via the documented decorators, which is why they're not part of the public Flask-Login api. Upstream packages are free to use them, the only thing that changed was making clear that they aren't part of the public api used by direct Flask-Login users.

**Comment by jwag956** [comment]
I can do that - however in general I don't like to use 'internal' parts of modules since of course you are free to change them at any time.
I don't understand the stmt 'use documented decorators' - the only 2 decorators check the results of authentication - they don't provide access to the authentication mechanisms (as implemented via the request_callback itself) themselves. 
Unless I am missing something this means upstream packages can't create their OWN decorators that still use  a login_manager instance.

What I am asking is - since as part of initialization I set these callbacks - can I, within the context of login_manager get them back?

**Comment by alanhamlett** [comment]
> can I, within the context of login_manager get them back?

We can add accessors to LoginManager like:

```python
@property
def request_callback(self):
    return self._request_callback
```

As long as we also add docs for those public methods.

**Comment by jwag956** [comment]
That would be great - exactly what I hoping for - it also would for example help some packages that are currently broken due to the change (less important IMHO).

**Comment by alanhamlett** [comment]
Actually, you want to set the callbacks not read them right? That's already possible from an upstream package by using the public methods, for ex: `login_manager.request_loader(custom_callback_fn)`.

@mattupstate I don't see Flask-Security reading the callback functions, only setting them. Probably because if the upstream sets a callback, they already have access to that callback and wouldn't need to read it from Flask-Login. Are there any use cases for needing to read/access the callback functions?

**Comment by jwag956** [comment]
Yes - I want to CALL the callback. and yes - Flask-Security does read/call the callback - look in decorators.py:_check_token:
`user = _security.login_manager.request_callback(request)`

The reason I want to (from my fork of Flask-Security) is that we allow the user to provide their OWN login_manager instance as part of setting up Flask-Security. So in the default case where Flask-Security creates its own login_manager and sets the callback - of course it can remember it. But in the case a user provides their own - we want to call theirs (in case they overrode flask-security's version). The best way to do that is to always use the initialized login_manager instance.

FYI - my fork is at:https://github.com/jwag956/flask-security

**Comment by alanhamlett** [comment]
Fixed with #472. @maxcountryman will need to release a new version with these 2 new properties.
